src/workspace-data/data-table/uri-viewer/UriViewer.js

```diff
@@ -90,9 +90,9 @@ export const UriViewer = _.flow(
   const renderTerminalCommand = (metadata) => {
     const { bucket, name } = metadata;
     const gsUri = `gs://${bucket}/${name}`;
-    const azUri = `${uri}?${metadata.sasToken}`;
+    // for public azure storage links, the sas token will be undefined
     const downloadCommand = isAzureUri(uri)
-      ? getDownloadCommand(metadata.name, azUri, metadata.accessUrl)
+      ? getDownloadCommand(metadata.name, metadata.sasToken ? `${uri}?${metadata.sasToken}` : uri, metadata.accessUrl)
```

Comment:
Could we use the same logic to determine the URL in both places?

Here, the URL shown is `metadata.sasToken ? `${uri}?${metadata.sasToken}` : uri`. For the download button, it's `metadata.azureSasStorageUrl || uri`.